src/test/java/picard/sam/RevertSamTest.java

```diff
@@ -592,4 +606,189 @@ public void testHardClipRoundTrip() throws Exception {
             }
         }
     }
+
+
+    private final static boolean OUTPUT_BY_READ_GROUP = true;
+
+    // Specifies the location of the dynamically created output map
+    private enum OutputMapOptions { LOCAL_OUTPUT_MAP,
+        CLOUD_OUTPUT_MAP,
+        NO_OUTPUT_MAP }
+
+
+    @DataProvider(name = "cloudTestData")
+    public Object[][] getCloudTestData() {
+        final PicardHtsPath tempCloudCram = PicardBucketUtils.getTempFilePath(DEFAULT_CLOUD_TEST_OUTPUT_DIR, "RevertSam", ".cram");
+        final PicardHtsPath tempLocalCram = PicardBucketUtils.getLocalTempFilePath("RevertSam", ".cram");
+        return new Object[][]{
+                // Output by read group without the output map, write output bams in the cloud.
+                {NA12878_MEDIUM, DEFAULT_CLOUD_TEST_OUTPUT_DIR, OUTPUT_BY_READ_GROUP, OutputMapOptions.NO_OUTPUT_MAP, null },
```

Comment:
For this test case, I think you can just create a temporary parent bucket in the staging area to to hold the output, rather than referencing the default staging bucket here and trying to synthesize unique names using a prefix and timestamp (see my comment where the name synthesis code is).

Reply:
I addressed this issue by "creating" a "directory" with a randomly generated path. See PicardBucketUtils::getRandomGCSDirectory()